source/main.c

```diff
@@ -1,11 +1,170 @@
 // main.c 
 // folder home
 #include <stdio.h>
-int main(int argc, char const *argv[])
+#include <stdlib.h>
+ 
+typedef struct
 {
-    printf("hello world \n");
-    getchar();
-
+	// Initialize front and rear 
+	int rear, front;
+	// Circular Queue 
+	int size;
+	int* arr;
+}Queue;
+ 
+void create( int s, Queue* q)
+{
+	q->front = -1;
+	q->rear = -1;
+	q->size = s;
+	//maloc
+	int array[s];
+	if(((q->arr =(int*)malloc(sizeof(array)))) == NULL)
+	{
+		printf("Allocation error.");
+		exit(0);
+	}
+}
+ 
+/* Function to create Circular queue */
+void enQueue(int value, Queue* q)
+{
+	if ((q->front == 0 && q->rear == q->size - 1) ||
+		(q->rear == (q->front - 1) % (q->size - 1)))
+	{
+		printf("\nQueue is Full");
+		return;
+	}
+ 
+	else if (q->front == -1) /* Insert First Element */
+	{
+		q->front = 0;
+		q->rear = 0;
+		*(q->arr + q->rear) = value;
+	}
+ 
+	else if (q->rear == q->size - 1 && q->front != 0)
+	{
+		q->rear = 0;
+		*(q->arr + q->rear) = value;
+	}
+ 
+	else
+	{
+		q->rear++;
+		*(q->arr + q->rear) = value;
+	}
+}
+ 
+// Function to delete element from Circular Queue 
+int deQueue(Queue* q)
+{
+	if (q->front == -1)
+	{
+		printf("\nQueue is Empty");
+		return NULL;
+	}
+ 
+	int data = *(q->arr + q->front);
+	*(q->arr + q->front) = -1;
+	if (q->front == q->rear)
+	{
+		q->front = -1;
+		q->rear = -1;
+	}
+	else if (q->front == q->size - 1)
+		q->front = 0;
+	else
+		q->front++;
+ 
+	return data;
+}
+ 
+// Function displaying the elements 
+// of Circular Queue 
+void displayQueue(Queue* q)
+{
+	if (q->front == -1)
+	{
+		printf("\nQueue is Empty");
+		return;
+	}
+	printf("\nElements in Circular Queue are: ");
+	if (q->rear >= q->front)
+	{
+		for (int i = q->front; i <= q->rear; i++)
+			printf("%d ", *(q->arr + i));
+	}
+	else
+	{
+		for (int i = q->front; i < q->size; i++)
+			printf("%d ", *(q->arr + i));
+ 
+		for (int i = 0; i <= q->rear; i++)
+			printf("%d ", *(q->arr + i));
+	}
+}
 
-    return 0;
+void delete (Queue *q)
+{
+	free(q->arr);
 }
+ 
+/* Driver of the program */
+int main()
+{
+	Queue q;
+	Queue* pointer = &q;
```

Comment:
It looks like an extra pointer, use &second_q instead everywhere.

Reply:
done